modules/arcore/arcore_interface.h

```diff
@@ -0,0 +1,117 @@
+/*************************************************************************/
```

Comment:
@BastiaanOlij What's the reason for ARCore being a module instead of an external plugin?

Reply:
With OpenVR/Oculus and Oculus mobile we don't have a choice, we rely on external libraries that need to be deployed if these are used that we can't deliver as part of the core application.

With ARCore we're using an API that is part of the OS so the functionality is available, this lowers the barrier of entry a fare amount. It really feels like unnecessary overkill to make this a plugin.

Also when I started it became clear that on iOS you can't have dynamic libraries (I started with ARKit) and at the time GDNative was pretty new, so there were still to many roadblocks. Today, yeah maybe part of this could be placed in a plugin. Whether there is benefit to it? I don't know, like I said I feel its overkill.

Reply:
@BastiaanOlij I'm about 3/4 through cleaning up the PR for compatibility with the new Android plugin system and I've got some bad and good news:

- Bad news: This will need to be an external plugin instead of a module. Unlike ARKit, the ARCore API is **not** part of the OS and require a [binary download](https://developers.google.com/ar/develop/c/enable-arcore#dependencies) for access. Asides from the license issues involved with shipping a closed source binary as part of the Godot Engine binary, this will unnecessarily increase the engine binary size, which given the size of the target audience for this feature, does not seem worth it.

- Good news: Looking at the logic here, this should all be doable by leveraging the gdnative-cpp bindings.

Reply:
@m4gr3d I think the only thing that needs work to move this into a gdnative plugin is that the camera server integration has a few holes. But we can plug those. 

I'm about to get some sleep but let's pick this up through email and see how easily we can transfer the code to a plugin. Seeing most of the relevant code is currently in a module and just an implementation of the ARVRInterface I suspect that most of this will be pretty straight forward especially after all the work we've done for the Quest.

Reply:
To me the "bad news" of having to make this an external plugin is a great news :)
The legal aspect of having to distribute a proprietary binary was definitely a big blocker for this PR, which was only made as a module as it seemed to be the only/most straightforward option at the time (but this would have likely still prevented actually merging it in core).

So having this moved to a GDNative plugin sounds great. Would it then be an Android plugin (using the new plugin system) that includes the GDNative library?

Reply:
@akien-mga Yes it would be an Android plugin with a GDNative library similar to how the Quest plugin is structured.